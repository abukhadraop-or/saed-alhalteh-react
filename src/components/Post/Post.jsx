// import React, { useEffect, useState } from 'react';
import React from 'react';
// import { useRouteMatch, useHistory } from 'react-router-dom';
import { useLocation } from 'react-router-dom';
import MetaInfo from 'components/MetaInfo/MetaInfo';
import LikeComponent from 'components/Like/Like';
// import { getPost } from 'services/fakePostsService';
import { HeaderContainer, PostContentContainer, Container, CommentsContainer } from './post.styles';

function Post() {
  // const [post, setPost] = useState({});
  // const match = useRouteMatch();
  const location = useLocation();
  const postToBeRendered = location.state.detail;

  // useEffect(() => {
  //   setPost(getPost(match.params.id));
  // }, []);

  return (
    <>
      <HeaderContainer>
        <h1>{postToBeRendered.title}</h1>
        <Container>
          <MetaInfo author={postToBeRendered.author} />
          <LikeComponent count={postToBeRendered.favoritesCount} />
        </Container>
      </HeaderContainer>
      <PostContentContainer>{postToBeRendered.body}</PostContentContainer>
      <CommentsContainer>
        Comments section will only be available for registered users ....
      </CommentsContainer>
    </>
  );
}

export default Post;
