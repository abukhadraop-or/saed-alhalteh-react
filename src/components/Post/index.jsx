import React from 'react';

/**
 * Post content.
 *
 * @return {JSX.Element} Post component.
 */
function Post() {
  return <h1>Post details </h1>;
}

export default Post;
