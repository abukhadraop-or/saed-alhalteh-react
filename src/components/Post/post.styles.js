import styled from 'styled-components';
import { colors } from 'shared/styles';

export const HeaderContainer = styled.div`
  color: ${colors.white};
  background-color: ${colors.mineShaft};
  padding: 15px 45px;
  & h1 {
    font-size: 2.8rem;
    font-weight: 600;
  }
`;

export const PostContentContainer = styled.div`
  padding: 25px 45px;
  font-size: 1.25rem;
  line-height: 1.5;
`;

export const Container = styled.div`
  display: flex;
`;
export const CommentsContainer = styled.div`
  padding: 25px 45px;
  font-size: 1.25rem;
  line-height: 1.5;
  color: ${colors.gray};
`;
