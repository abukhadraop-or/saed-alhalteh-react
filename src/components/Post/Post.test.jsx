import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Post from '.';

describe('<Post /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Post />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
