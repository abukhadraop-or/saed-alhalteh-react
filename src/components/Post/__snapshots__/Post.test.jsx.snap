// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> tests matches the snapshot 1`] = `
<h1>
  Post details 
</h1>
`;
